src/libraries/Pool.sol

```diff
@@ -326,7 +325,7 @@ library Pool {
 
         SwapCache memory cache = SwapCache({
             liquidityStart: self.liquidity,
-            protocolFee: params.zeroForOne ? uint8(slot0Start.protocolFee % 256) : uint8(slot0Start.protocolFee >> 8)
+            protocolFee: params.zeroForOne ? uint16(slot0Start.protocolFee % 4096) : uint16(slot0Start.protocolFee >> 12)
```

Comment:
Maybe add helpers in the fee library to extract the 1->0 and 0->1 fees?

```
slot0Start.protocolFee.toZeroForOne();
slot0Start.protocolFee.toOneForZero();
```

Reply:
cause you could use them also in the `_isFeeWithinBounds` function

Reply:
& then also would be nice/easier to write unit tests making sure that the 0 to 1 and 1 to 0 extraction are as expected

Reply:
we dont have a Fees library anymore, only SwapFeeLibrary...I could make a ProtocolFeeLibrary?

Reply:
sure that works, just more so meant to put it in a helper function/lib so you can reuse it since this math is done in a few places